core/Plugin/Metric.php

```diff
@@ -119,18 +119,30 @@ public static function getMetric($row, $columnName, $mappingNameToId = null)
             ) {
                 $value = $row->getColumn($mappingNameToId[$columnName]);
             }
-        } else {
-            $value = @$row[$columnName];
-            if ($value === null
-                && isset($mappingNameToId[$columnName])
-            ) {
-                $columnName = $mappingNameToId[$columnName];
-                $value = @$row[$columnName];
-            }
+
             return $value;
+
+        } elseif (!empty($row)) {
+            
+            if (array_key_exists($columnName, $row)) {
```

Comment:
Not really related to the pull request but cleaned it up when I noticed that `@` was used. Will be also faster in case the array key does not exist => performance improvement.